### PR TITLE
Add vertical offset to osu! playfield to match stable

### LINKED ISSUE #9457: Storyboard is not aligned perfectly to the playfield
**Describe the bug:** 
circle size works differently in lazer than in stable, causing issues with maps that use storyboard for fake circles (disconnected hardkore, aleph-0, storyboard-based aspire maps)

**Screenshots or videos showing encountered issue:** 
Screen Shot 2020-07-08 at 12 09 03 PM
From Aspire V's "RiraN - Unshakable". The slider that the cursor is on is real, as is the one fading in on the left, but the one in the bottom right is not.

**osu!lazer version:** 
2020.707.0

### PR DESCRIPTION
**Pull request by Wieku**
Fixes #9457

### CODE REVIEW

File: osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs [hunk 1]
```diff
@@ -55,6 +55,7 @@ protected override void Update()
                 // Scale = 819.2 / 512
                 // Scale = 1.6
                 Scale = new Vector2(Parent.ChildSize.X / OsuPlayfield.BASE_SIZE.X);
+                Position = new Vector2(0, 8 * Parent.ChildSize.Y / OsuPlayfield.BASE_SIZE.Y);
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: This should probably have a disable for the editor, as stable does. Exposing it as a property seems to be the easiest way to do so and then setting it in `DrawableOsuEditRuleset`.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs (hunk 1 above)
Comment (Wieku) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: I completely forgot that this also affects the editor.
Do you mean as a `PlayfieldShift` property with `8` set as default which is then set to 0 by `DrawableOsuEditRuleset`?

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: Yeah, that would be fine.

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: for future proofing, how would this work with the design mode in editor?

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: Does design mode turn this off somehow? Or is the exclusion for the editor just because of the menu bar (in which case this probably shouldn't exist)?

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: as in we want the storyboard to align correctly in design mode too. so i'm not sure it should ever be turned off?

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: Yeah that makes sense. Let's go with this as is then.

Comment (Wieku) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: So maybe better solution is to use bool `PlayfieldShift` as a property? It will be better than passing arbitrary values to the object. I think we want to turn on shift in design mode because aligning storyboard objects to hitobjects in osu-stable editor is a huge pain.

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: If we're looking to fix historically awkward offsets, this could potentially be applied at the legacy storyboard parsing step (moving all elements to their correct location) rather than adding local offsets.

I'd ask that we don't move too quickly on merging this and actually get it right.

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: That was brought up in the original thread, with the issue that it creates a black border at the bottom of the screen.

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: Am I missing something? stable applies this offset to gameplay too. So if we apply the same offset here, and change nothing about the editor and keep the offset, then things will align correctly and the saved positions would be post-offset so that gameplay should still match up when imported into stable?

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:58]: if we are comfortable with keeping the offset as-is going forward then yes, applying at the playfield is the correct solution. given it's something like 12 "gameplay" pixels i'm not sure we want to, but maybe more effort to break compatibility than keep it at this point?

**Comment by Wieku** [comment]
I made it to be only turned on in DrawableOsuRuleset because leaving this always on lead to misaligned compose tab:
osu_2020-08-03_13-40-21

**Comment by bdach** [comment]
Not sure doing this just for the osu! ruleset is correct. Wouldn't this mean that the storyboards would remain visually broken when playing converts in other game modes?

**Comment by Wieku** [comment]
This mostly affects maps which storyboards are made with osu! ruleset in mind so regardless of that shift those storyboards would be broken on converts anyway.

File: osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs [hunk 2]
```diff
@@ -55,6 +65,7 @@ protected override void Update()
                 // Scale = 819.2 / 512
                 // Scale = 1.6
                 Scale = new Vector2(Parent.ChildSize.X / OsuPlayfield.BASE_SIZE.X);
+                Position = new Vector2(0, (PlayfieldShift ? 8f : 0f) * Parent.ChildSize.Y / OsuPlayfield.BASE_SIZE.Y);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:68]: Not a huge fan of this being done in Update but I guess it's in line with what is already there.

File: osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs [hunk 3]
```diff
@@ -15,6 +15,14 @@ public class OsuPlayfieldAdjustmentContainer : PlayfieldAdjustmentContainer
 
         private const float playfield_size_adjust = 0.8f;
 
+        /// <summary>
+        /// Whether an osu-stable playfield offset should be applied (8 osu!pixels)
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:19]: This should mention the reasoning for applying the offset (ie. for storyboard alignment).

Also we have not used the term "osu!pixels" anywhere else so please don't introduce it here. The value doesn't even need to be mentioned in the xmldoc because the description of what it does should be enough to satisfy someone setting this externally.

**Review by peppy — DISMISSED** [review]

File: osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs (hunk 3 above)
Comment (Wieku) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:19]: So reword it to something like this?
```
Whether an osu-stable playfield offset should be applied.
Set to true if correct alignment with storyboards is needed.
```

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:19]: I'd rename the property to `AlignWithStoryboard`.

"When true, an offset is applied to allow alignment with historical storyboards displayed in the same parent space. This will shift the playfield downwards slightly."

Comment (Wieku) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfieldAdjustmentContainer.cs:19]: Yeah, that sounds much better

**Review by bdach — APPROVED** [review]
seems good :+1:

**Review by smoogipoo — APPROVED** [review]
lgtm
